dotnet/samples/KernelSyntaxExamples/Example00_01_PlayFabDataQnA.cs

```diff
@@ -434,25 +474,290 @@ private async Task<string> CreateAndExcecutePythonScript(string question, string
+    {
+        this._titleId = titleId ?? throw new ArgumentNullException(nameof(titleId));
+        this._cosmosClient = new(
+            endpointUrl ?? throw new ArgumentNullException(nameof(endpointUrl)),
+            primaryKey ?? throw new ArgumentNullException(nameof(primaryKey)));
+
+        this._databaseName = databaseName ?? throw new ArgumentNullException(nameof(databaseName));
+        this._containerName = containerName ?? throw new ArgumentNullException(nameof(containerName));
+    }
+
+    public async Task<GameReport?> FetchAsync(string documentId, CancellationToken cancellationToken)
     {
-        // Remove from the beginning
-        while (source.StartsWith(wordToRemove, StringComparison.OrdinalIgnoreCase))
+        // Get a reference to the database and container
+        Database database = this._cosmosClient.GetDatabase(this._databaseName);
+        Microsoft.Azure.Cosmos.Container container = database.GetContainer(this._containerName);
+
+        try
+        {
+            // Read the document by its ID
+            ItemResponse<GameReport> response = await container.ReadItemAsync<GameReport>(documentId, new PartitionKey(this._titleId), cancellationToken: cancellationToken);
+            return response.Resource;
+        }
+        catch (CosmosException ex)
         {
-            source = source.Substring(wordToRemove.Length);
+            Console.WriteLine($"Error: {ex.Message}");
+            return null;
         }
+    }
 
-        // Remove from the end
-        while (source.EndsWith(wordToRemove, StringComparison.OrdinalIgnoreCase))
+    public async Task<IList<GameReport>> FetchByQueryAsync(string query, CancellationToken cancellationToken)
+    {
+        // Get a reference to the database and container
+        Database database = this._cosmosClient.GetDatabase(this._databaseName);
+        Microsoft.Azure.Cosmos.Container container = database.GetContainer(this._containerName);
+
+        List<GameReport> ret = new();
+        try
+        {
+            // Execute the query
+            QueryDefinition queryDefinition = new(query);
+            FeedIterator<dynamic> resultSetIterator = container.GetItemQueryIterator<dynamic>(queryDefinition);
+
+            while (resultSetIterator.HasMoreResults)
+            {
+                FeedResponse<dynamic> response = await resultSetIterator.ReadNextAsync(cancellationToken);
+                foreach (dynamic item in response)
+                {
+                    GameReport? gameReport = ((JObject)item).ToObject<GameReport>();
+                    if (gameReport is not null)
+                    {
+                        ret.Add(gameReport);
+                    }
+                }
+            }
+        }
+        catch (CosmosException ex)
         {
-            source = source.Substring(0, source.Length - wordToRemove.Length);
+            Console.WriteLine($"Error: {ex.Message}");
+            return new List<GameReport>();
```

Comment:
Can we return something to the user, when there is no data? After the exception, there is no data, so we can't proceed to next steps.

Reply:
Make sense, I'll just let it throw in this case